recipes-kernel/linux/files/patches-6.1/0036-arm64-dts-ti-iot2050-Add-icssg-prueth-nodes-for-PG1-.patch

```diff
@@ -0,0 +1,184 @@
+From: Jan Kiszka <jan.kiszka@siemens.com>
+Date: Sat, 27 Jun 2020 10:53:13 +0200
+Subject: [PATCH 36/69] arm64: dts: ti: iot2050: Add icssg-prueth nodes for PG1
+ and PG2 devices
+
+Add the required nodes to enable ICSSG SR1.0 and SR2.0 based prueth
+networking.
+
+Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
+---
+ .../boot/dts/ti/k3-am65-iot2050-common.dtsi   | 129 +++++++++++++++++-
+ 1 file changed, 128 insertions(+), 1 deletion(-)
+
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
+index 180bfb2a9ddf..1be99de1cae2 100644
+--- a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
+@@ -11,12 +11,15 @@
+ 
+ #include "k3-am654.dtsi"
+ #include <dt-bindings/phy/phy.h>
++#include <dt-bindings/net/ti-dp83867.h>
+ 
+ / {
+ 	aliases {
+ 		spi0 = &mcu_spi0;
+ 		mmc0 = &sdhci1;
+ 		mmc1 = &sdhci0;
++		ethernet1 = &icssg0_emac0;
++		ethernet2 = &icssg0_emac1;
+ 	};
+ 
+ 	chosen {
+@@ -102,6 +105,80 @@ dp_refclk: clock {
+ 		#clock-cells = <0>;
+ 		clock-frequency = <19200000>;
+ 	};
++
++	/* Dual Ethernet application node on PRU-ICSSG0 */
++	icssg0_eth: icssg0-eth {
++		compatible = "ti,am654-icssg-prueth";
++		pinctrl-names = "default";
++		pinctrl-0 = <&icssg0_rgmii_pins_default>;
++		sram = <&msmc_ram>;
++
++		ti,prus = <&pru0_0>, <&rtu0_0>, <&tx_pru0_0>,
++			<&pru0_1>, <&rtu0_1>, <&tx_pru0_1>;
++		firmware-name = "ti-pruss/am65x-sr2-pru0-prueth-fw.elf",
++				"ti-pruss/am65x-sr2-rtu0-prueth-fw.elf",
++				"ti-pruss/am65x-sr2-txpru0-prueth-fw.elf",
++				"ti-pruss/am65x-sr2-pru1-prueth-fw.elf",
++				"ti-pruss/am65x-sr2-rtu1-prueth-fw.elf",
++				"ti-pruss/am65x-sr2-txpru1-prueth-fw.elf";
++
++		ti,pruss-gp-mux-sel = <2>,      /* MII mode */
++				      <2>,
++				      <2>,
++				      <2>,	/* MII mode */
++				      <2>,
++				      <2>;
++
++		ti,mii-g-rt = <&icssg0_mii_g_rt>;
++		ti,mii-rt = <&icssg0_mii_rt>;
++		ti,iep = <&icssg0_iep0>,  <&icssg0_iep1>;
++
++		interrupt-parent = <&icssg0_intc>;
++		interrupts = <24 0 2>, <25 1 3>;
++		interrupt-names = "tx_ts0", "tx_ts1";
++
++		dmas = <&main_udmap 0xc100>, /* egress slice 0 */
++		       <&main_udmap 0xc101>, /* egress slice 0 */
++		       <&main_udmap 0xc102>, /* egress slice 0 */
++		       <&main_udmap 0xc103>, /* egress slice 0 */
++		       <&main_udmap 0xc104>, /* egress slice 1 */
++		       <&main_udmap 0xc105>, /* egress slice 1 */
++		       <&main_udmap 0xc106>, /* egress slice 1 */
++		       <&main_udmap 0xc107>, /* egress slice 1 */
++
++		       <&main_udmap 0x4100>, /* ingress slice 0 */
++		       <&main_udmap 0x4101>, /* ingress slice 1 */
++		       <&main_udmap 0x4102>, /* mgmnt rsp slice 0 */
++		       <&main_udmap 0x4103>; /* mgmnt rsp slice 1 */
++		dma-names = "tx0-0", "tx0-1", "tx0-2", "tx0-3",
++			    "tx1-0", "tx1-1", "tx1-2", "tx1-3",
++			    "rx0", "rx1",
++			    "rxmgm0", "rxmgm1";
```

Comment:
`"rxmgm0", "rxmgm1"` is gone in ti-linux-6.1.y, k3-am654-base-board.dts. Likely just a style difference.

Reply:
But looking that the DTS again, that delta in the ti-linux rather seems like a bug there. Let's keep things as-is.